lua/nvim-tree/explorer/filters.lua

```diff
@@ -69,6 +69,30 @@ local function dotfile(path)
   return M.config.filter_dotfiles and utils.path_basename(path):sub(1, 1) == "."
 end
 
+---@param path string
+---@return boolean
+local function binary(path)
```

Comment:
I'm sorry, we can't include this OS specific functionality. It will also be a big performance overhead.

Fortunately we have `:help executable()` that is portable and fast. That is persisted on the Node.

Reply:
I did not know of this function, thanks! :)

Reply:
One downside to this option is that executable text files, such as shell scripts, will be hidden with this filter.

Reply:
I see. That is a good use case, however as you say it is a good case for a freely defined user function.